src/tools/wpa/DataModel/QuicConnection.cs

```diff
@@ -0,0 +1,234 @@
+                    {
+                        break;
+                    }
+                case QuicEventId.ConnAssignWorker:
+                    {
+                        if (Worker != null)
+                        {
+                            Worker.OnConnectionRemoved();
+                        }
+                        var payload = evt.Payload as QuicConnectionAssignWorkerPayload;
+                        var key = new QuicObjectKey(evt.PointerSize, payload!.WorkerPointer, evt.ProcessId);
+                        Worker = state.FindOrCreateWorker(key);
+                        Worker.OnConnectionAdded();
+                        break;
+                    }
+                case QuicEventId.ConnTransportShutdown:
+                    {
+                        var payload = evt.Payload as QuicConnectionTransportShutdownPayload;
+                        State = QuicConnectionState.Shutdown;
+                        IsAppShutdown = false;
+                        IsShutdownRemote = payload!.IsRemoteShutdown != 0;
+                        ShutdownTimeStamp = evt.TimeStamp;
+                        break;
+                    }
+                case QuicEventId.ConnAppShutdown:
+                    {
+                        var payload = evt.Payload as QuicConnectionAppShutdownPayload;
+                        State = QuicConnectionState.Shutdown;
+                        IsAppShutdown = true;
+                        IsShutdownRemote = payload!.IsRemoteShutdown != 0;
+                        ShutdownTimeStamp = evt.TimeStamp;
+                        break;
+                    }
+                case QuicEventId.ConnHandleClosed:
+                    {
+                        State = QuicConnectionState.Closed;
+                        break;
+                    }
+                case QuicEventId.ConnOutFlowStats:
+                    {
+                        var payload = evt.Payload as QuicConnectionOutFlowStatsPayload;
+                        BytesSent = payload!.BytesSent;
+                        break;
+                    }
+                case QuicEventId.ConnOutFlowBlocked:
+                    {
+                        break;
+                    }
+                case QuicEventId.ConnInFlowStats:
+                    {
+                        var payload = evt.Payload as QuicConnectionInFlowStatsPayload;
+                        BytesReceived = payload!.BytesRecv;
+                        break;
+                    }
+                case QuicEventId.ConnStats:
+                    {
+                        var payload = evt.Payload as QuicConnectionStatsPayload;
+                        BytesSent = payload!.SendTotalBytes;
+                        BytesReceived = payload!.RecvTotalBytes;
+                        break;
+                    }
+                default:
+                    break;
+            }
+
+            FinalTimeStamp = evt.TimeStamp;
+
+            if (Worker != null)
```

Comment:
You don't need to change this unless you want to, but this call and others that you want to either null check or ignore can be done like `Worker?.OnConnectionEvent(evt);` The `?` will do a null check, and it does it a bit faster too. In multi threaded contexts it much more important, but since we're not its fine, and its just a slight cleanup if you want it in the future.

Reply:
Ok. I will make the changes.